packages/backend/src/services/ServiceRepository.ts

```diff
@@ -1 +1,204 @@
+
+/**
+ * Interface outlining all services available under the `ServiceRepository`. Add new services to
+ * this list (in alphabetical order, please!) to have typescript help ensure you've updated the
+ * service repository correctly.
+ */
+interface ServiceManifest {
+    analyticsService: AnalyticsService;
+    commentService: CommentService;
+    csvService: CsvService;
+    dashboardService: DashboardService;
+    downloadFileService: DownloadFileService;
+    encryptionService: EncryptionService;
+    gitIntegrationService: GitIntegrationService;
+    githubAppService: GithubAppService;
+    gdriveService: GdriveService;
+    groupService: GroupsService;
+    healthService: HealthService;
+    notificationService: NotificationsService;
+    organizationService: OrganizationService;
+    personalAccessTokenService: PersonalAccessTokenService;
+    pinningService: PinningService;
+    projectService: ProjectService;
+    savedChartService: SavedChartService;
+    schedulerService: SchedulerService;
+    searchService: SearchService;
+    shareService: ShareService;
+    sshKeyPairService: SshKeyPairService;
+    spaceService: SpaceService;
+    unfurlService: UnfurlService;
+    userAttributesService: UserAttributesService;
+    userService: UserService;
+    validationService: ValidationService;
+}
+
+/**
+ * Enforces the presence of getter methods for all services declared in the manifest.
+ */
+type ServiceFactoryMethod<T extends ServiceManifest> = {
+    [K in keyof T as `get${Capitalize<string & K>}`]: () => T[K];
+};
+
+/**
+ * Intermediate abstract class used to enforce service factory methods via the `ServiceFactoryMethod`
+ * type. We need this extra thin layer to ensure we are statically aware of all members.
+ */
+abstract class ServiceRepositoryBase {
+    /**
+     * Container for service instances. Can be replaced with bare class members once
+     * this class is handling service instantiation directly, and not just behaving as
+     * a dumb proxy.
+     */
+    protected _services: ServiceManifest;
+
+    constructor({ services }: { services: ServiceManifest }) {
+        this._services = services;
+    }
+}
+
+/**
+ * Bare service repository class, which acts as a container for all existing
+ * services, and as a point to share instantiation and common logic.
+ *
+ * If you need to access a service, you should do it through an instance of this
+ * repository - ideally one that you accessed through a controller, or otherwise
+ * via dependency injection.
+ *
+ * NOTE: For now, this repository simply exposes services instantiated in `./services.ts`,
+ *       and provided to this repository directly. At a later stage, this repository will
+ *       handle instantiating all services internally, including cross-service dependencies.
+ */ export class ServiceRepository
+    extends ServiceRepositoryBase
+    implements ServiceFactoryMethod<ServiceManifest>
+{
+    public getAnalyticsService(): AnalyticsService {
```

Comment:
This whole section for now is pretty pointless, until `ServiceRepository` is actually instantiating services. The individual methods allow us to have fine control over how each service is instantiated, and for e.g test mocks or overrides to extend `ServiceRepository` and override service getters.